src/test/kotlin/uk/gov/dwp/dataworks/egress/services/impl/DataServiceImplTest.kt

```diff
@@ -170,9 +171,85 @@ class S3ServiceImplTest: WordSpec() {
                 verify(compressionService, times(100)).compress(any(), any())
                 verifyNoMoreInteractions(compressionService)
             }
+
```

Comment:
Needs a test for unsupported transfer type somewhere. Can just be a unit test here I think.